### ISSUE TITLE
DetailRowTrigger on DataGrid is called twice on each row click

### ISSUE THREAD

**Issue by GilShalit** [Megabit/Blazorise#2500]
Using the following to close a details row if the row is clicked a second time
```
<DataGrid TItem="CourtCase"
		  @bind-SelectedRow="@selectedCourtCase"
		  DetailRowTrigger="@((item)=> item.Id == selectedCourtCase?.Id && !isRowAlreadySelected())">
...
</DataGrid>
```

with
```
bool isRowAlreadySelected()
        {
            bool isRowClicked = prevSelectdRow!=null && prevSelectdRow.Id == selectedCourtCase.Id;
            prevSelectdRow = selectedCourtCase;
            return isRowClicked;
        }
```

Does not work.  Using debug I found that isRowAlreadySelected() is called twice on each row click, so obviously the prevSelectedRow variable is already set to the row which is newly selected on the second call, and so the result is effectively always true.

I found a way around this with an additional flag, but I think this isn't how it should work.

**Comment by stsrki** [comment]
I would say this is good behavior. DetailRowTrigger can be potentially called for every row in a table. That is why it has `item` as an argument. It's not just for the currently selected row.

**Comment by GilShalit** [comment]
Right.  But for some reason it seems to be called twice with the same item and selectedCourtCase because the first part of the trigger condition is met twice so that isRowAlreadySelected() is visited twice for each row click.

**Comment by StephenOTT** [comment]
Having similar issues: This seems to fit into the Virtualization work for the Datagrid?

**Comment by David-Moreira** [comment]
The detail row should close with Ctrl+click, have you guys tried that? Or does that not satisfy your use case?

**Comment by GilShalit** [comment]
Yes it works with Ctrl+Click, but this does not seem as intuitive as simply have the second click close an opened row.  How do I know it is not intuitive? EVERY new user asks me why isn't the second click closing the opened row...

**Comment by stsrki** [comment]
@David-Moreira do you think we should change the toggle with a click instead of ctrl+click?

**Comment by David-Moreira** [comment]
Well, @GilShalit does have a point. I would say most users, would just expect to just click back again and it would collapse the `DetailRowTrigger` .
However the reason why Ctrl+Click works, is because it will unselect the row, therefore the `DetailRowTrigger Func` will not evaluate to true anymore.

I think it's feasible.
At the moment `DetailRowTrigger` is evaluated on a per blazor state update basis. Maybe we could start evaluating it only on the actual clicks, that should also solve it being possibly triggered more than once due to blazor state updates.
- Note: Need to check how it behaves with MultiSelectAll and if we actually do need it to evaluate on blazor state updates for some reason.

@stsrki let me know your thoughts.

**Comment by stsrki** [comment]
@David-Moreira I didn't try it but multi-select might be problematic indeed.

I will assign it to you. Also, I have placed in the new **Last Minute** milestone. It means, in case we got enough time before the 0.9.4 release you can try to do it.
